File: ImapX.Sample/JsonHelper.cs
using System;
using System.IO;
using System.Runtime.Serialization.Json;
using System.Text;

namespace ImapX.Sample
{
    public class JsonHelper
    {
        public static string To<T>(T obj)
        {
            var serializer = new DataContractJsonSerializer(obj.GetType());
            using (var ms = new MemoryStream())
            {
                serializer.WriteObject(ms, obj);
                string retVal = Encoding.Default.GetString(ms.ToArray());
                return retVal;
            }
        }

        public static T From<T>(string json)
        {
            var obj = Activator.CreateInstance<T>();
            using (var ms = new MemoryStream(Encoding.Unicode.GetBytes(json)))
            {
                var serializer = new DataContractJsonSerializer(obj.GetType());
                obj = (T) serializer.ReadObject(ms);
            }

            return obj;
        }
    }
}

File: ImapX.VBSample/JsonHelper.vb
Imports System.IO
Imports System.Text
Imports System.Runtime.Serialization.Json

Public Class JsonHelper

    Public Shared Function From(Of T)(ByVal json As String) As T
        Dim obj As T = Activator.CreateInstance(Of T)()
        Using ms As MemoryStream = New MemoryStream(Encoding.Unicode.GetBytes(json))
            Dim serializer As New DataContractJsonSerializer(obj.GetType)
            Return DirectCast(serializer.ReadObject(ms), T)
        End Using
    End Function

    Public Shared Function [To](Of T)(ByVal obj As T) As String
        Dim serializer As New DataContractJsonSerializer(obj.GetType)
        Using ms As MemoryStream = New MemoryStream
            serializer.WriteObject(ms, obj)
            Return Encoding.Default.GetString(ms.ToArray)
        End Using
    End Function

End Class
